compliance/evidence.py

```diff
@@ -81,22 +79,38 @@ def from_locker(cls, path, locker):
         path = substitute_config(path)
         return cls(locker.get_evidence(path))
 
+    @property
+    def rootdir(self):
+        raise NotImplementedError('The root directory is not set.')
+
+    @property
+    def dir_path(self):
+        return str(PurePath(self.rootdir).joinpath(self.category))
+
     @property
     def name(self):
-        return self._name + '.' + self.extension
+        return substitute_config(self._name)
 
     @property
     def path(self):
-        return os.path.join(self.dir_path, self.name)
+        return str(PurePath(self.dir_path).joinpath(self.name))
 
     @property
-    def dir_path(self):
-        return os.path.join(self.rootdir, self.category)
+    def extension(self):
+        return PurePath(self.name).suffix.lstrip('.')
 
     @property
     def content(self):
         return self._content
 
+    @property
+    def content_as_json(self):
+        if self.extension != 'json':
+            return
```

Comment:
Perhaps raising a `ValueError` might be more deterministic.

Reply:
Added in https://github.com/ComplianceAsCode/auditree-framework/pull/80/commits/6c71069dd93a05230ce429fda162668c9576183a